dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/WorkflowItemRestRepository.java

```diff
@@ -294,4 +317,30 @@ protected void delete(Context context, Integer id) {
             throw new RuntimeException(e.getMessage(), e);
         }
     }
+
+    /**
+     * Checks if @link{SUBMIT_EDIT_METADATA} is a valid option in the workflow step this task is currently at.
+     * Patching and uploading is only allowed if this is the case.
+     * @param context               Context
+     * @param xmlWorkflowItem       WorkflowItem of the task
+     */
+    private void checkIfEditMetadataAllowedInCurrentStep(Context context, XmlWorkflowItem xmlWorkflowItem) {
+        try {
+            ClaimedTask claimedTask = claimedTaskService.findByWorkflowIdAndEPerson(context, xmlWorkflowItem,
+                context.getCurrentUser());
+            if (claimedTask == null) {
+                throw new UnprocessableEntityException("WorkflowItem with id " + xmlWorkflowItem.getID()
+                    + " has not been claimed yet.");
+            }
+            Workflow workflow = workflowFactory.getWorkflow(claimedTask.getWorkflowItem().getCollection());
+            Step step = workflow.getStep(claimedTask.getStepID());
+            WorkflowActionConfig currentActionConfig = step.getActionConfig(claimedTask.getActionID());
+            if (!currentActionConfig.getProcessingAction().getOptions().contains(SUBMIT_EDIT_METADATA)) {
+                throw new UnprocessableEntityException(SUBMIT_EDIT_METADATA + " is not a valid option on this " +
+                    "action (" + currentActionConfig.getProcessingAction().getClass() + ").");
+            }
+        } catch (SQLException | WorkflowConfigurationException e) {
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
This should provide a more useful error message or just be updated to `new RuntimeException(e);`  Simply returning `e.getMessage()` isn't useful here.

Reply:
All generic exceptions in this repo given relevant messages